support/Environments/setup/frontera_gcc.sh

```diff
@@ -0,0 +1,200 @@
+    tar -xzf yaml-cpp.tar.gz
+    mv yaml-cpp-* yaml-cpp-build
+    cd $dep_dir/yaml-cpp-build
+    mkdir build
+    cd build
+    cmake -D CMAKE_BUILD_TYPE=Release -D YAML_CPP_BUILD_TESTS=OFF \
+          -D CMAKE_C_COMPILER=gcc -D CMAKE_CXX_COMPILER=g++ \
+          -D YAML_CPP_BUILD_CONTRIB=OFF \
+          -D YAML_CPP_BUILD_TOOLS=ON \
+          -D CMAKE_INSTALL_PREFIX=$dep_dir/yaml-cpp ..
+    make -j4
+    make install
+    cd $dep_dir
+    rm -r yaml-cpp-build
+    rm -r yaml-cpp.tar.gz
+    echo "Installed yaml-cpp into $dep_dir/yaml-cpp"
+    cat >$dep_dir/modules/yaml-cpp <<EOF
+#%Module1.0
+prepend-path LIBRARY_PATH "$dep_dir/yaml-cpp/lib"
+prepend-path LD_LIBRARY_PATH "$dep_dir/yaml-cpp/lib"
+prepend-path CPLUS_INCLUDE_PATH "$dep_dir/yaml-cpp/include"
+prepend-path CMAKE_PREFIX_PATH "$dep_dir/yaml-cpp/"
+EOF
+fi
+cd $dep_dir
+
+if [ -f $dep_dir/libsharp/lib/libsharp.a ]; then
+    echo "libsharp is already installed"
+else
+    echo "Installing libsharp..."
+    wget https://github.com/Libsharp/libsharp/archive/v1.0.0.tar.gz -O libsharp.tar.gz
+    tar -xzf libsharp.tar.gz
+    mv libsharp-* libsharp_build
+    cd $dep_dir/libsharp_build
+    autoconf
+    ./configure --prefix=$dep_dir/libsharp --disable-openmp
+    make -j4
+    mv ./auto $dep_dir/libsharp
+    cd $dep_dir
+    rm -r libsharp_build
+    rm libsharp.tar.gz
+    echo "Installed libsharp into $dep_dir/libsharp"
+    cat >$dep_dir/modules/libsharp <<EOF
+#%Module1.0
+prepend-path LIBRARY_PATH "$dep_dir/libsharp/lib"
+prepend-path LD_LIBRARY_PATH "$dep_dir/libsharp/lib"
+prepend-path CPLUS_INCLUDE_PATH "$dep_dir/libsharp/include"
+prepend-path CMAKE_PREFIX_PATH "$dep_dir/libsharp/"
+EOF
+fi
+cd $dep_dir
+
+# Set up Charm++ because that can be difficult
+charm_config=ucx-linux-x86_64-smp
+if [ -f $dep_dir/charm/${charm_config}/lib/libck.a ]; then
+    echo "Charm++ is already installed"
+else
+    echo "Installing Charm++..."
+    wget https://github.com/UIUC-PPL/charm/archive/v6.10.2.tar.gz
+    tar xzf v6.10.2.tar.gz
+    mv charm-6.10.2 charm
+    cd $dep_dir/charm
+    ./build LIBS ${charm_config} --with-production -j6
+    cd $dep_dir
+    rm v6.10.2.tar.gz
+    echo "Installed Charm++ into $dep_dir/charm"
+    cat >$dep_dir/modules/charm <<EOF
+#%Module1.0
+prepend-path PATH "$dep_dir/charm/${charm_config}/bin"
+prepend-path LIBRARY_PATH "$dep_dir/charm/${charm_config}/lib"
+prepend-path LD_LIBRARY_PATH "$dep_dir/charm/${charm_config}/lib"
+prepend-path CPLUS_INCLUDE_PATH "$dep_dir/charm/${charm_config}/include"
+prepend-path CMAKE_PREFIX_PATH "$dep_dir/charm/${charm_config}/"
+setenv CHARM_VERSION 6.8.2
+setenv CHARM_HOME $dep_dir/charm/${charm_config}
+setenv CHARM_ROOT $dep_dir/charm/${charm_config}
+EOF
+fi
+cd $dep_dir
+
+if [ -f $dep_dir/modules/spectre_boost ]; then
+    echo "Boost is already set up"
+else
+    cat >$dep_dir/modules/spectre_boost <<EOF
+#%Module1.0
+prepend-path CPLUS_INCLUDE_PATH "$BOOST_ROOT/include"
+EOF
+fi
+cd $dep_dir
+
```

Comment:
Re your question on `h5py`: I suggest you install Py packages in the `$dep_dir` like this:

```sh
python3 -m venv --prompt spectre --system-site-packages $dep_dir/py_env
source $dep_dir/py_env/bin/activate
# Install h5py with the system HDF5
HDF5_DIR=$TACC_HDF5_DIR pip install --no-binary=h5py h5py
# Install additional Python packages. User can install more using `pip` if they
# want.
pip install yapf==0.29.0
```

Instead of sourcing the `activate` script you can add things to the `PATH` manually if you prefer, e.g. in a module. The `activate` script does only very little.